server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryService.java

```diff
@@ -308,10 +297,18 @@ private boolean mergeDeprecatedComponentParameters(DbSession session, Boolean on
     return effectiveOnComponentOnly;
   }
 
-  private void failIfBothParametersSet(@Nullable Collection<String> uuids, @Nullable Collection<String> keys, String message) {
-    if (uuids != null && keys != null) {
-      throw new IllegalArgumentException(message);
+  private static boolean atMostOneElementNotNull(@Nullable Object... objects) {
+    int counter = 0;
+    for (Object object : objects) {
```

Comment:
Using guava, you could do something like : 
from(object).filter(Predicates.noNull()).anyMatch(new MatchMoreThanOneElement(counter));